Use EnumSet for BoxSides

anttijk opened the pull request:
#### 3dd2b46fd2d83abdd41f1ee7670252b3684d6fd7
<pre>
Use EnumSet for BoxSides
<a href="https://bugs.webkit.org/show_bug.cgi?id=300680">https://bugs.webkit.org/show_bug.cgi?id=300680</a>
<a href="https://rdar.apple.com/162581907">rdar://162581907</a>

Reviewed by Tim Nguyen.

Eliminate the now-unnecessary BoxSideFlag enum.
We can also remove various conversion functions as a result.

* Source/WebCore/page/LocalFrameView.cpp:
(WebCore::LocalFrameView::fixedContainerEdges const):
* Source/WebCore/page/Page.cpp:
(WebCore::Page::updateFixedContainerEdges):
* Source/WebCore/page/Page.h:
* Source/WebCore/platform/BoxSides.h:
(WebCore::boxAxisToFlag): Deleted.

Also remoe BoxAxisFlag enum.

(WebCore::boxSideFromFlag): Deleted.
* Source/WebCore/platform/FixedContainerEdges.cpp:
(WebCore::FixedContainerEdges::fixedEdges const):
* Source/WebCore/rendering/BorderEdge.h:
(WebCore::edgesShareColor):
(WebCore::includesAdjacentEdges):
(WebCore::edgeFlagForSide): Deleted.
(WebCore::includesEdge): Deleted.
* Source/WebCore/rendering/BorderPainter.cpp:
(WebCore::BorderPainter::paintSides const):
(WebCore::BorderPainter::paintTranslucentBorderSides const):
(WebCore::borderStyleHasUnmatchedColorsAtCorner):
(WebCore::BorderPainter::paintBorderSides const):
* Source/WebCore/rendering/RenderBoxModelObject.h:
* Source/WebCore/rendering/style/RenderStyle.h:
* Source/WebCore/rendering/style/RenderStyleInlines.h:
(WebCore::RenderStyle::anchorFunctionScrollCompensatedAxes const):
* Source/WebCore/rendering/style/RenderStyleSetters.h:
(WebCore::RenderStyle::setAnchorFunctionScrollCompensatedAxes):
* Source/WebCore/rendering/style/StyleRareNonInheritedData.h:
* Source/WebCore/style/AnchorPositionEvaluator.cpp:
(WebCore::AnchorScrollAdjuster::AnchorScrollAdjuster):
(WebCore::Style::AnchorPositionEvaluator::addAnchorFunctionScrollCompensatedAxis):
* Source/WebKit/UIProcess/API/ios/WKWebViewIOS.mm:
(-[WKWebView _setObscuredInsetsInternal:]):
* Source/WebKit/WebProcess/WebPage/Cocoa/WebPageCocoa.mm:
(WebKit::WebPage::sidesRequiringFixedContainerEdges const):

Canonical link: <a href="https://commits.webkit.org/301472@main">https://commits.webkit.org/301472@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/72c3fdf8671387ecfc3b2d7dc129d555f920ca46

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/126084 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/45737 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/36523 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/132896 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/77887 "Built successfully") | ⏳ 🛠 ios-apple 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/127955 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/46408 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/54282 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/96032 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [❌ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/64117 "") | ⏳ 🛠 mac-apple 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/129032 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/37130 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/112765 "Passed tests") | [❌ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/76510 "1 api test failed or timed out") | | ⏳ 🛠 vision-apple 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/36032 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/30947 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/76368 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/106912 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/31169 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/135603 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/52840 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/40573 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/104525 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/53298 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/108982 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/104238 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/26559 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/49633 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/27961 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/50223 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/52737 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/58571 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/52065 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/55411 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/53775 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

nt1m commented on Source/WebCore/platform/BoxSides.h:
```diff
@@ -44,27 +45,11 @@ enum class LogicalBoxAxis : uint8_t {
-enum class BoxAxisFlag : uint8_t {
-    Horizontal = 1 << 0,
-    Vertical = 1 << 1
+    HighestEnumValue = Vertical
```
Curious where we use `HighestEnumValue`. I couldn't find references in enumset.h

anttijk replied:
> // If the enum has a member called HighestEnumValue it will be used to compute the storage size. Otherwise 64 bits are used.

anttijk replied:
you should be looking at EnumSet.h in WTF

nt1m commented on Source/WebCore/platform/BoxSides.h:
```diff
@@ -93,14 +78,9 @@ enum class BoxSide : uint8_t {
-    Right   = 1 << static_cast<unsigned>(BoxSide::Right),
-    Bottom  = 1 << static_cast<unsigned>(BoxSide::Bottom),
-    Left    = 1 << static_cast<unsigned>(BoxSide::Left)
+    HighestEnumValue = Left
```
ditto

nt1m commented on Source/WebCore/rendering/BorderPainter.cpp:
```diff
@@ -628,10 +628,10 @@ static inline bool borderStyleHasUnmatchedColorsAtCorner(BorderStyle style, BoxS
-        BoxSideSet topRightSides = { BoxSideFlag::Top, BoxSideFlag::Right };
-        BoxSideSet bottomLeftSides = { BoxSideFlag::Bottom, BoxSideFlag::Left };
+        BoxSideSet topRightSides = { BoxSide::Top, BoxSide::Right };
+        BoxSideSet bottomLeftSides = { BoxSide::Bottom, BoxSide::Left };
 
-        BoxSideSet usedSides { edgeFlagForSide(side), edgeFlagForSide(adjacentSide) };
+        BoxSideSet usedSides { side, adjacentSide };
```
very tempting to rewrite this with `auto`

anttijk committed:
Use EnumSet for BoxSides
https://bugs.webkit.org/show_bug.cgi?id=300680
rdar://162581907

Reviewed by Tim Nguyen.

Eliminate the now-unnecessary BoxSideFlag enum.
We can also remove various conversion functions as a result.

* Source/WebCore/page/LocalFrameView.cpp:
(WebCore::LocalFrameView::fixedContainerEdges const):
* Source/WebCore/page/Page.cpp:
(WebCore::Page::updateFixedContainerEdges):
* Source/WebCore/page/Page.h:
* Source/WebCore/platform/BoxSides.h:
(WebCore::boxAxisToFlag): Deleted.

Also remoe BoxAxisFlag enum.

(WebCore::boxSideFromFlag): Deleted.
* Source/WebCore/platform/FixedContainerEdges.cpp:
(WebCore::FixedContainerEdges::fixedEdges const):
* Source/WebCore/rendering/BorderEdge.h:
(WebCore::edgesShareColor):
(WebCore::includesAdjacentEdges):
(WebCore::edgeFlagForSide): Deleted.
(WebCore::includesEdge): Deleted.
* Source/WebCore/rendering/BorderPainter.cpp:
(WebCore::BorderPainter::paintSides const):
(WebCore::BorderPainter::paintTranslucentBorderSides const):
(WebCore::borderStyleHasUnmatchedColorsAtCorner):
(WebCore::BorderPainter::paintBorderSides const):
* Source/WebCore/rendering/RenderBoxModelObject.h:
* Source/WebCore/rendering/style/RenderStyle.h:
* Source/WebCore/rendering/style/RenderStyleInlines.h:
(WebCore::RenderStyle::anchorFunctionScrollCompensatedAxes const):
* Source/WebCore/rendering/style/RenderStyleSetters.h:
(WebCore::RenderStyle::setAnchorFunctionScrollCompensatedAxes):
* Source/WebCore/rendering/style/StyleRareNonInheritedData.h:
* Source/WebCore/style/AnchorPositionEvaluator.cpp:
(WebCore::AnchorScrollAdjuster::AnchorScrollAdjuster):
(WebCore::Style::AnchorPositionEvaluator::addAnchorFunctionScrollCompensatedAxis):
* Source/WebKit/UIProcess/API/ios/WKWebViewIOS.mm:
(-[WKWebView _setObscuredInsetsInternal:]):
* Source/WebKit/WebProcess/WebPage/Cocoa/WebPageCocoa.mm:
(WebKit::WebPage::sidesRequiringFixedContainerEdges const):

Canonical link: https://commits.webkit.org/301472@main
